resolver/src/p2p/peer/actor.rs

```diff
@@ -0,0 +1,303 @@
+use super::{
+    config::Config,
+    fetcher::Fetcher,
+    ingress::{Mailbox, Message},
+};
+use crate::{
+    p2p::{
+        wire::{self, peer_msg::Payload},
+        Director, Producer, Value,
+    },
+    Consumer,
+};
+use commonware_cryptography::{Array, Scheme};
+use commonware_macros::select;
+use commonware_p2p::utils::requester::Requester;
+use commonware_p2p::{Receiver, Recipients, Sender};
+use commonware_runtime::{Clock, Spawner};
+use commonware_utils::futures::Pool as FuturesPool;
+use futures::{
+    channel::{mpsc, oneshot},
+    future::{self, Either},
+    StreamExt,
+};
+use governor::{
+    clock::Clock as GClock, middleware::NoOpMiddleware, state::keyed::HashMapStateStore,
+    RateLimiter,
+};
+use prost::Message as _;
+use rand::Rng;
+use std::marker::PhantomData;
+use tracing::{debug, error, warn};
+
+/// An actor that makes and responds to requests using the P2P network.
+pub struct Actor<
+    E: Clock + GClock + Spawner + Rng,
+    C: Scheme,
+    D: Director<PublicKey = C::PublicKey>,
+    Key: Array,
+    Con: Consumer<Key = Key, Value = Value, FailureCode = ()>,
+    Pro: Producer<Key = Key, Value = Value>,
+    NetS: Sender<PublicKey = C::PublicKey>,
+    NetR: Receiver<PublicKey = C::PublicKey>,
+> {
+    ////////////////////////////////////////
+    // Interfaces
+    ////////////////////////////////////////
+    runtime: E,
+    consumer: Con,
+    producer: Pro,
+    director: D,
+
+    ////////////////////////////////////////
+    // Outgoing Requests
+    ////////////////////////////////////////
+    /// Mailbox that makes and cancels fetch requests
+    mailbox: mpsc::Receiver<Message<Key>>,
+
+    fetcher: Fetcher<E, C, Key, NetS>,
+
+    ////////////////////////////////////////
+    // Incoming Requests
+    ////////////////////////////////////////
+    serves: FuturesPool<(C::PublicKey, u64, Result<Value, oneshot::Canceled>)>,
+    serve_concurrent: usize,
+
+    // Rate limit for incoming requests per peer
+    #[allow(clippy::type_complexity)]
+    rate_limiter:
```

Comment:
This rate limiter is unnecessary as we already handle rate limiting on the provided network connection?

That will ensure we don't get too many messages from a single PublicKey 👍